src/providers/on-chain-quote-provider.ts

```diff
@@ -1223,6 +1223,60 @@ export class OnChainQuoteProvider implements IOnChainQuoteProvider {
       } attempt loops. Total calls made to provider: ${totalCallsMade}. Have retried for timeout: ${haveRetriedForTimeout}`
     );
 
+    // Log total routes
+    metric.putMetric(
+      `${this.metricsPrefix(
+        this.chainId,
+        useMixedRouteQuoter,
+        mixedRouteContainsV4Pool,
+        protocol,
+        optimisticCachedRoutes
+      )}RoutesLength`,
+      routesQuotes.length,
+      MetricLoggerUnit.Count
+    );
+
+    // Log total quotes
+    metric.putMetric(
+      `${this.metricsPrefix(
+        this.chainId,
+        useMixedRouteQuoter,
+        mixedRouteContainsV4Pool,
+        protocol,
+        optimisticCachedRoutes
+      )}RoutesQuotesLength`,
+      routesQuotes
+        .map((routeWithQuotes) => routeWithQuotes[1].length)
```

Comment:
hmm... total quotes can be simplified to success quotes + failed quotes right? so that we dont have to repeat the logic from `flatMap((routeWithQuotes: RouteWithQuotes<TRoute>) => routeWithQuotes[1])`